cmscontrib/PrometheusExporter.py

```diff
@@ -0,0 +1,293 @@
+        data = data.all()
+        metric.add_metric([], data[0][0])
+        yield metric
+
+    def collect_users(self, session):
+        metric = GaugeMetricFamily(
+            "cms_participations",
+            "Number of participations grouped by category",
+            labels=["category"],
+        )
+        base_data = session.query(func.count(distinct(Participation.id))) \
+            .select_from(Participation)
+        if self.contest_id is not None:
+            base_data = base_data.filter(Participation.contest_id == self.contest_id)
+
+        data = base_data.all()
+        metric.add_metric(["total"], data[0][0])
+
+        data = base_data.filter(Participation.hidden == True).all()
+        metric.add_metric(["hidden"], data[0][0])
+
+        data = base_data.filter(Participation.unrestricted == True).all()
+        metric.add_metric(["unrestricted"], data[0][0])
+
+        data = base_data.filter(Participation.starting_time != None).all()
+        metric.add_metric(["started"], data[0][0])
+
+        data = base_data.join(Submission).all()
+        metric.add_metric(["submitted"], data[0][0])
+
+        data = base_data \
+            .join(Submission) \
+            .join(SubmissionResult) \
+            .join(Dataset) \
+            .join(Task, Dataset.task_id == Task.id) \
+            .filter(Task.active_dataset_id == SubmissionResult.dataset_id) \
+            .filter(SubmissionResult.score > 0) \
+            .all()
+        metric.add_metric(["non_zero"], data[0][0])
+
+        yield metric
+
+
+def main():
+    """Parse arguments and launch process."""
+    parser = argparse.ArgumentParser(description="Prometheus exporter.")
+    parser.add_argument(
+        "--host",
+        help="IP address to bind",
+        default="0.0.0.0",
+    )
+    parser.add_argument(
+        "--port",
+        help="Port to use",
+        default=8811,
+        type=int,
+    )
+    parser.add_argument(
+        "--contest-id",
+        "-c",
+        help="Obtain metrics for the specified contest only",
+        type=int,
+    )
+    parser.add_argument(
+        "--no-submissions",
+        help="Do not export submissions metrics",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--no-workers",
+        help="Do not export workers metrics",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--no-queue",
+        help="Do not export queue metrics",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--no-communications",
+        help="Do not export communications metrics",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--no-users",
+        help="Do not export users metrics",
+        action="store_true",
+    )
+
+    args = parser.parse_args()
+
+    service = PrometheusExporter(args)
+    REGISTRY.register(service)
+    start_http_server(args.port, addr=args.host)
+    print("Started at http://%s:%s/metric" % (args.host, args.port))
```

Comment:
This code could be moved to a new method in `PrometheusExporter`. In fact, `PrometheusExporter.run()` could also be overridden and this new method be called from there.